cmd/agent/agent.go

```diff
@@ -246,6 +248,19 @@ func loop(c *cli.Context) error {
 	return nil
 }
 
+func runWithRetry(context *cli.Context) error {
+	var err error
+	for i := 0; i < retryCount; i++ {
+		if err = run(context); err == nil {
+			break
+		} else if status.Code(err) == codes.Unavailable {
+			log.Warn().Err(err).Msg(fmt.Sprintf("cannot connect to server, retrying in %v", retryDelay))
+			time.Sleep(retryDelay)
+		}
```

Comment:
if err != nil and no rpc error we should break and return err too

Reply:
Sorry, looks like my brain was not working well. Fixed.